conftest.py

```diff
@@ -3,10 +3,36 @@
 import requests
 from time import sleep
 import logging
+from dataclasses import dataclass
+
 
 log = logging.getLogger(__name__)
 
 
+@dataclass
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int
+
+    def __post_init__(self):
+        
+        for item in ['name', 'hid', 'model']:
+
+            if not isinstance(getattr(self, item), str):
+                raise TypeError(f"'{item}' should be a string")
+
+            if getattr(self, item) == "":
+                raise ValueError(f"'{item}' should not be empty")
+            
+        for item in ['firmware_version', 'reading_interval']:
```

Comment:
We check string arguments for value errors, so we can do the same to these two as well.

We'd like to test this for validity from sensor's standpoint. That would be checking that the firmware version is between 10 and 15, both ends included. You could test this with the following code:
```
10 <= self.firmware_version <= 15
```
For reading interval we're interested in checking that this is also a valid value from sensor's standpoint. That would be seeing that it's equal or more than 1.